apps/channels/datamodels.py

```diff
@@ -46,48 +47,47 @@ def parse(update_obj) -> "TelegramMessage":
 
 class TwilioMessage(BaseModel):
     """
-    A wrapper class for user messages coming from the whatsapp
+    A wrapper class for user messages coming from the twilio
     """
 
-    from_number: str
-    to_number: str
+    from_: str
+    to: str
     body: str
     content_type: MESSAGE_TYPES | None = Field(default=MESSAGE_TYPES.TEXT)
+    platform: ChannelPlatform
     media_url: str | None = Field(default=None)
     content_type_unparsed: str | None = Field(default=None)
 
-    @field_validator("to_number", "from_number", mode="before")
-    @classmethod
-    def strip_prefix(cls, value):
-        return value.split("whatsapp:")[1]
-
     @field_validator("content_type", mode="before")
     @classmethod
     def determine_content_type(cls, value):
         if not value:
             # Normal test messages doesn't have a content type
             return MESSAGE_TYPES.TEXT
-        if value and value == "audio/ogg":
+        if value and value in ["audio/ogg", "video/mp4"]:
             return MESSAGE_TYPES.VOICE
 
     @property
     def chat_id(self) -> str:
-        return self.from_number
+        return self.from_
 
     @property
     def message_text(self) -> str:
         return self.body
 
     @staticmethod
     def parse(message_data: dict) -> "TwilioMessage":
+        prefix_channel_map = {"messenger": ChannelPlatform.FACEBOOK, "whatsapp": ChannelPlatform.WHATSAPP}
+        prefix = message_data["From"].split(":")[0]
         content_type = message_data.get("MediaContentType0")
         return TwilioMessage(
-            from_number=message_data["From"],
-            to_number=message_data["To"],
+            from_=message_data["From"].split(f"{prefix}:")[1],
+            to=message_data["To"].split(f"{prefix}:")[1],
```

Comment:
Rather than `.split()[1]` you should use `.removeprefix(...)` which is simpler and safer

Reply:
https://github.com/dimagi/open-chat-studio/pull/341/commits/42cae86d8f3cc0bf5a9638afe61f24b16d3653b8